### PR TITLE
[browser][MT] enable MT PLINQ

### PR DESCRIPTION
**Pull request by pavelsavara**
Use normal threads in MT build of PLINQ rather than 
- `SinglePartitionMode => false;` when `FEATURE_WASM_THREADS`
- enabled more unit tests, which are no longer failing. There are still few which deadlock or have trouble.

### CODE REVIEW

**Review by ilonatommy — APPROVED** [review]
Looks logical

File: src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.cs [hunk 1]
```diff
@@ -238,8 +238,10 @@ public static void Invoke(ParallelOptions parallelOptions, params Action[] actio
             {
                 // If we've gotten this far, it's time to process the actions.
 
+#if !FEATURE_WASM_THREADS
                 // Web browsers need special treatment that is implemented in TaskReplicator
                 if (OperatingSystem.IsBrowser() ||
+#endif
```
Comment (campersau) [inline src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.cs:244]: This looks wrong. There is no `if (` when `FEATURE_WASM_THREADS` is enabled.

File: src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <StartupObject />
+    <_WasmPThreadPoolSize Condition="'$(MonoWasmBuildVariant)' == 'multithread'">32</_WasmPThreadPoolSize>
```
Comment (kg) [inline src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj:8]: A value this high may cause reliability issues

Comment (pavelsavara) [reply src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj:8]: this is temporary and not production code

Comment (pavelsavara) [reply src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj:8]: If we have deputy design, we could let emscripten to allocate pthreads on the fly. 
That said these unit tests are nasty about creating many threads.

**Comment by pavelsavara** [comment]
image

File: src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj [hunk 3]
```diff
@@ -6,6 +6,9 @@
   <PropertyGroup>
     <StartupObject />
   </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetOS)' == 'browser'">
+    <WasmXHarnessMonoArgs>--setenv=XHARNESS_LOG_TEST_START=true --no-memory-snapshot</WasmXHarnessMonoArgs>
```
Comment (kg) [inline src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj:10]: maybe add a comment explaining why this is here so we know whether we can remove it later

**Review by kg — APPROVED** [review]

File: src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj (hunk 3 above)
Comment (pavelsavara) [reply src/libraries/System.Threading.Tasks.Parallel/tests/System.Threading.Tasks.Parallel.Tests.csproj]: I remove it, it helps you to see which test was running last and deadlocked.

**Pull request merged**